### ISSUE TITLE
Inconsistent ProcessPoolExecutor behaviour on macOS between 3.7 and 3.8/9

### ISSUE THREAD

**Issue by e7368eed-17dc-41b1-9b33-511902d7fe58** [python/cpython#86447]
BPO | [42281](https://bugs.python.org/issue42281)
--- | :---
Nosy | @brianquinlan, @ronaldoussoren, @pitrou, @ned-deily, @mustafaquraish
Files | <li>[test.py](https://bugs.python.org/file49579/test.py "Uploaded as text/plain at 2020-11-07.00:19:50 by @mustafaquraish"): Sample code with a bug</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2020-11-07.00:19:51.420>
labels = ['OS-mac', 'type-bug', '3.8']
title = 'Inconsistent ProcessPoolExecutor behaviour on macOS between 3.7 and 3.8/9'
updated_at = <Date 2020-11-07.09:28:38.301>
user = 'https://github.com/mustafaquraish'
```

bugs.python.org fields:
```python
activity = <Date 2020-11-07.09:28:38.301>
actor = 'ronaldoussoren'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['macOS']
creation = <Date 2020-11-07.00:19:51.420>
creator = 'mustafaquraish'
dependencies = []
files = ['49579']
hgrepos = []
issue_num = 42281
keywords = []
message_count = 2.0
messages = ['380483', '380495']
nosy_count = 5.0
nosy_names = ['bquinlan', 'ronaldoussoren', 'pitrou', 'ned.deily', 'mustafaquraish']
pr_nums = []
priority = 'normal'
resolution = 'wont fix'
stage = 'resolved'
status = 'pending'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue42281'
versions = ['Python 3.8']
```

</p></details>

**Comment by zitterbewegung** [comment]
Python 3.7, 3.8 and 3.9 are only accepting security updates so this issue should be closed.

**Comment by ronaldoussoren** [comment]
I agree that this issue can be closed, but not because 3.7-9 are only accepting security updates:  This issue is not a bug, but due to a explicit change of behaviour of multiprocessing in 3.8 (as mentioned in my previous message from 2020).
